devlib/module/thermal.py

```diff
@@ -93,15 +154,44 @@ def __init__(self, target):
                 continue
 
             if re_match.group(1) == 'thermal_zone':
-                self.add_thermal_zone(re_match.group(2))
+                self._add_thermal_zone(re_match.group(2))
             elif re_match.group(1) == 'cooling_device':
                 # TODO
                 pass
 
-    def add_thermal_zone(self, _id):
+    def _add_thermal_zone(self, _id):
         self.zones[int(_id)] = ThermalZone(self.target, self.thermal_root, _id)
 
     def disable_all_zones(self):
         """Disables all the thermal zones in the target"""
         for zone in self.zones.values():
             zone.set_enabled(False)
+
+    @asyn.asyncf
+    async def get_all_temperatures(self, error='raise'):
+        """
+        Returns dictionary with current reading of all thermal zones.
+
+        :params error: Sensor read error handling (raise or ignore)
+        :type error: str
+
+        :returns: a dictionary in the form: {tz_type:temperature}
+        """
+
+        if not isinstance(error, str):
+            raise TypeError('error should a str')
+
+        async def get_temperature_noexcep(item):
+            tzid, tz = item
+            try:
+                temperature = await tz.get_temperature.asyn()
+            except TargetStableCalledProcessError as e:
+                if error != 'ignore':
+                    raise e
+                self.logger.warning('Skipping thermal_zone_id={} thermal_zone_type={} error="{}"'.format(tzid, tz.type, e))
+                return None
```

Comment:
This can be an exhaustive check:
```
if error == 'ignore':
    self.logger(...)
elif error == 'raise':
    raise
else:
   raise ValueError(f'Unknown error parameter value: {error}')
```
And you can remove the earlier check on error being a string.